embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/limits/UpToLimitStrategy.kt

```diff
@@ -0,0 +1,29 @@
+package io.embrace.android.embracesdk.arch.limits
+
+import io.embrace.android.embracesdk.internal.utils.Provider
+import io.embrace.android.embracesdk.logging.InternalEmbraceLogger
+import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger
+
+/**
+ * Allows capturing data up until a limit, then stops capturing.
+ */
+internal class UpToLimitStrategy(
```

Comment:
Do we at least need threadsafety, if not synchronization, at this level?

Reply:
Added synchronisation around a lock.